include/boost/astronomy/time/parser.hpp

```diff
@@ -0,0 +1,66 @@
+/*=============================================================================
+Copyright 2020 Syed Ali Hasan <alihasan9922@gmail.com>
+
+Distributed under the Boost Software License, Version 1.0. (See accompanying
+file License.txt or copy at https://www.boost.org/LICENSE_1_0.txt)
+=============================================================================*/
+
+#ifndef BOOST_ASTRONOMY_PARSER_HPP
+#define BOOST_ASTRONOMY_PARSER_HPP
+
+#include <string>
+#include <iostream>
+
+using namespace std;
+
+struct decimal_hour {
+private:
+    double dh = 0;
+
+    double time = 0;
+    int hours = 0;
+    double minutesRemainder = 0;
+    int minutes = 0;
+    double secondsRemainder = 0;
+    int seconds = 0;
+
+public:
+ decimal_hour(double const& d){
+      dh = d;
+
+      time = dh;
+      hours = (int)time;
+      minutesRemainder = (time - hours) * 60;
+      minutes = (int)minutesRemainder;
+      secondsRemainder = (minutesRemainder - minutes) * 60;
+      seconds = (int)secondsRemainder;
+    }
+
+    decimal_hour(double const& h,double const& m,double const& s){
+      dh = h + m/60 + s/(60*60);
+    }
+
+    double get() const{
+        return (double)dh;
```

Comment:
why to return `double` with it is always going to be `long` or `'int` in this case?

Reply:
Yeah, no reason to typecast. Will remove it.